cmd/chainbridge/deploycontracts.go

```diff
@@ -0,0 +1,216 @@
+
+	}
+
+	return validatorAddresses
+}
+
+func updateNonce(auth *bind.TransactOpts, client *ethclient.Client, deployAddress common.Address) (*bind.TransactOpts, error) {
+
+	newNonce, err := client.PendingNonceAt(context.Background(), deployAddress)
+	if err != nil {
+		log.Error("error fetching latest nonce")
+		return nil, err
+	}
+
+	auth.Nonce = big.NewInt(int64(newNonce))
+
+	return auth, nil
+
+}
+
+func accountSetUp(port string, validators int, deployPK string) (*ethclient.Client, *bind.TransactOpts, common.Address, []common.Address, error) {
+
+	client, err := ethclient.Dial("http://localhost:" + port)
+	if err != nil {
+		log.Error("error connecting to client")
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	privateKey, err := crypto.HexToECDSA(deployPK)
+	if err != nil {
+		log.Error("error casting private key to ECDSA")
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		log.Error("error casting public key to EDCSA")
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	deployAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+	nonce, err := client.PendingNonceAt(context.Background(), deployAddress)
+	if err != nil {
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	gasPrice, err := client.SuggestGasPrice(context.Background())
+	if err != nil {
+		log.Error("error fetching latest gas price")
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	auth := bind.NewKeyedTransactor(privateKey)
+	auth.Nonce = big.NewInt(int64(nonce))
+	auth.Value = big.NewInt(0)
+	auth.GasLimit = uint64(6721975)
+	auth.GasPrice = gasPrice
+
+	validatorAddresses := createValidatorSlice(VALIDATOR_ADDRESS, validators)
+
+	return client, auth, deployAddress, validatorAddresses, nil
+
+}
+
+func deployReceiver(auth *bind.TransactOpts, client *ethclient.Client, validatorAddresses []common.Address, depositThreshold *big.Int, validatorThreshold *big.Int, deployAddress common.Address) (common.Address, error) {
+
+	auth, err := updateNonce(auth, client, deployAddress)
+	if err != nil {
+		return ZERO_ADDRESS, err
+	}
+
+	recAddr, _, _, err := receiver.DeployReceiver(auth, client, validatorAddresses, depositThreshold, validatorThreshold)
+	if err != nil {
+		log.Error("error deploying reciever instance")
+		return ZERO_ADDRESS, err
+	}
+
+	return recAddr, nil
+}
+
+func deployEmitter(auth *bind.TransactOpts, client *ethclient.Client, deployAddress common.Address) (common.Address, error) {
+
+	auth, err := updateNonce(auth, client, deployAddress)
+	if err != nil {
+		return ZERO_ADDRESS, err
+	}
+
+	emitterAddr, _, _, err := emitter.DeployEmitter(auth, client)
+	if err != nil {
+		log.Error("error deploying emitter instance")
+		return ZERO_ADDRESS, err
+	}
+
+	return emitterAddr, nil
+
+}
+
+func deployBridgeAsset(auth *bind.TransactOpts, client *ethclient.Client, mc uint8, deployAddress common.Address) (common.Address, error) {
```

Comment:
Can we add a test to this? It could be dependent on ganache running (which i believe the existing tests already account for).

Reply:
What exactly are you looking to test?

Integrating with existing tests is a separate issue.